tests/model_explainability/lm_eval/utils.py

```diff
@@ -29,3 +30,20 @@ def verify_lmevaljob_running(client: DynamicClient, lmevaljob: LMEvalJob) -> Non
     lmevaljob_pod.wait_for_status(status=lmevaljob_pod.Status.RUNNING, timeout=Timeout.TIMEOUT_10MIN)
 
     check_pod_status_in_time(pod=lmevaljob_pod, status={lmevaljob_pod.Status.RUNNING, lmevaljob_pod.Status.SUCCEEDED})
+
+
+def wait_for_lmevaljob_state(
```

Comment:
Please use `@retry` decorator instead. It would make the function short and simple.